// @nehalist/gatsby-theme-nehalem/gatsby-config.js
module.exports = (themeOptions) => {
  const loadDefaultPages = themeOptions.loadDefaultPages !== undefined ? themeOptions.loadDefaultPages : true;
  const contentPath      = themeOptions.contentPath || 'content';
  const manifest         = themeOptions.manifest ? themeOptions.manifest : {
    name: `nehalem - A Gatsby theme`,
    short_name: `ACTROS`,
    start_url: `/`,
    background_color: `#a4cbb8`,
    theme_color: `#a4cbb8`,
    display: `minimal-ui`,
    icon: `${__dirname}/assets/intellrobotslab_icon.png`
  };

  return {
    siteMetadata: {
      title: `ACTROS`,
      siteUrl: `https://actros-educational-project.github.io`,
      description: `Center for Advanced Training on Robotics and %TOPICS%`,
      topics: [
        'Open Source',
        'Navigation',
        'Deep Learning',
        'Perception',
      ],
      menu: [
        {
          name: 'Home',
          path: '/'
        },
        /*{
          name: 'About',
          path: '/about'
        },*/
        {
          name: 'Courses',
          path: '/courses'
        },

        {
          name: 'Contact',
          path: '/contact'
        }
      ],
      footerMenu: [
        {
          name: 'Courses',
          path: '/courses'
        },
      ],
      search: true,
      author: {
        name: `ACTROS`,
        description: `The project <a href="https://www.rosin-project.eu/ftp/center-for-advanced-training-on-robotics-and-open-source-act-ros-education-project" target="_blank"><strong>“Center for Advanced Training on Robotics and Open Source (ACTROS)”</strong></a> aims to create a reference center in mobile robotics training using Open Source, mainly ROS and ROS2. This center is within the <a href="https://www.urjc.es/" target="_blank"><strong>Rey Juan Carlos University</strong></a>, which is currently a national benchmark in ROS training, both by university courses taught since 2015 (Advanced University Course in Robot Programming with ROS) and by the creation of a Degree in Software Robotics, which is a pioneer in our country because of its focus on software development for robots.
        This project aims to equip the existing laboratory robotics robots reference in the field (Pepper, Tiago, Kobuki, etc.) to carry out regularly run courses both for advanced students, and professionals who want to train in software development for robots. This project has received funding from the European Union’s Horizon 2020 research and innovation programme under grant agreement No. 732287.`,
        social: {
          facebook: ``,
          twitter: `https://twitter.com/IntellRobotLabs`,
          linkedin: ``,
          instagram: ``,
          youtube: ``,
          github: `https://github.com/ACTROS-Educational-Project`,
          twitch: ``
        }
      }
    },
    plugins: [
      `gatsby-plugin-typescript`,
      `gatsby-transformer-sharp`,
      `gatsby-plugin-react-helmet`,
      `gatsby-plugin-styled-components`,
      `gatsby-plugin-sitemap`,
      `gatsby-plugin-sharp`,
      {
        resolve: `gatsby-plugin-manifest`,
        options: manifest
      },
      {
        resolve: `gatsby-source-filesystem`,
        options: {
          name: 'content',
          path: contentPath
        }
      },
      {
        resolve: `gatsby-source-filesystem`,
        options: {
          name: `themeAssets`,
          path: `${__dirname}/assets`
        }
      },
      {
        resolve: `gatsby-transformer-yaml`,
        options: {
          typeName: `Tags`
        }
      },
      {
        resolve: `gatsby-plugin-lunr`,
        options: {
          languages: [
            {
              name: 'en'
            }
          ],
          fields: [
            { name: 'title', store: true, attributes: { boost: 20 } },
            { name: 'content', store: true },
            { name: 'tags', store: true },
            { name: 'excerpt', store: true },
            { name: 'path', store: true }
          ],
          resolvers: {
            MarkdownRemark: {
              title: node => node.frontmatter.title,
              content: node => node.html,
              tags: node => node.frontmatter.tags,
              excerpt: node => node.frontmatter.excerpt,
              path: node => node.frontmatter.path
            }
          }
        }
      },
      {
        resolve: `gatsby-transformer-remark`,
        options: {
          plugins: [
            `gatsby-remark-autolink-headers`,
            `gatsby-remark-prismjs`,
            {
              resolve: `gatsby-remark-images`,
              options: {
                maxWidth: 1200
              }
            }
          ]
        }
      },
      loadDefaultPages && {
        resolve: `gatsby-plugin-page-creator`,
        options: {
          path: `${__dirname}/src/pages`
        }
      },
      {
        resolve: `gatsby-plugin-feed`,
        options: {
          query: `
            {
              site {
                siteMetadata {
                  title
                  description
                  siteUrl
                  site_url: siteUrl
                }
              }
            }
          `,
          feeds: [
            {
              serialize: ({ query: { site, allMarkdownRemark } }) => {
                return allMarkdownRemark.edges.map(edge => {
                  return Object.assign({}, edge.node.frontmatter, {
                    description: edge.node.frontmatter.excerpt,
                    date: edge.node.frontmatter.created,
                    url: site.siteMetadata.siteUrl + edge.node.frontmatter.path,
                    guid: site.siteMetadata.siteUrl + edge.node.frontmatter.path,
                    custom_elements: [{ "content:encoded": edge.node.html }],
                  })
                })
              },
              query: `
              {
                allMarkdownRemark(
                  sort: { order: DESC, fields: [frontmatter___created] },
                  filter: { fileAbsolutePath: { regex: "/(posts)/.*\\\\.md$/" } }
                ) {
                  edges {
                    node {
                      html
                      frontmatter {
                        title
                        excerpt
                        path
                        created
                      }
                    }
                  }
                }
              }
              `,
              output: `/rss.xml`,
              title: `RSS Feed`
            }
          ]
        }
      }
    ].filter(Boolean)
  };
};
